src/Entity/Enemy.cpp

```diff
@@ -0,0 +1,159 @@
+								CollisionComponent& collisionComponent, VelocityComponent& velocityComponent)
+{
+	Vector2 playerScreenPosition, enemyScreenPosition;
+	//QUESTION: how would i save the player ref on the init stage? Would it be another component as well? 
+	playerScreenPosition = enemyEntity.world().component<CharacterTag>().target<CharacterTag>().get_ref<TransformComponent>()->mPosition;
+	enemyScreenPosition = transformComponent.mPosition;
+	Vector2 oldDirection = velocityComponent.mDirection;
+	
+	velocityComponent.mDirection = Vector2Subtract(playerScreenPosition, enemyScreenPosition);
+
+	//TODO: get collision working here properly and get dimention from the collision object instead of 50
+	if (Vector2Length(velocityComponent.mDirection) < 50.0f)
+	{
+		animationStateComponent.mCurrentAnimName = AnimationName::IDLE;
+		velocityComponent.mDirection = {};
+		return;
+	}
+
+	//Changing animation row (aka switching to different animation) when running
+	animationStateComponent.mCurrentAnimName = AnimationName::RUN;
+
+	//Flipping orientation of the character if direction sign flipped as well
+	if (oldDirection.x != velocityComponent.mDirection.x)
+	{
+		transformComponent.mScale.x = (velocityComponent.mDirection.x < 0.0f) ? -fabs(transformComponent.mScale.x) : fabs(transformComponent.mScale.x);
+	}
+
+	//Check for map bounds
+	const flecs::entity mapEntity = enemyEntity.world().component<MapTag>().target<MapTag>();
+	const Texture2D mapTexture = mapEntity.get_ref<TextureComponent>()->mTexture;
+
+	const Vector2 newPosition = Vector2Add(transformComponent.mPosition, 
+											Vector2Scale(Vector2Normalize(velocityComponent.mDirection), velocityComponent.mSpeed * GetFrameTime()));
+
+	// Check if the new position is within map boundaries
+	if (newPosition.x >= 0 && newPosition.x <= mapTexture.width - ENEMY_WIDTH &&
+		newPosition.y >= 0 && newPosition.y <= mapTexture.height - ENEMY_HEIGHT)
+	{
+		const Rectangle newEntityRect{newPosition.x, newPosition.y, collisionComponent.mRectScale.x, collisionComponent.mRectScale.y};
+
+		bool hasCollided{false};
+		Rectangle otherRect{};
+		auto filter = enemyEntity.world().filter<CollisionComponent, TransformComponent>();
+
+		filter.each([&hasCollided, &newEntityRect, &otherRect, &enemyEntity](flecs::entity e, CollisionComponent& otherCollisionComponent, TransformComponent& otherTransformComponent) {
+			// Avoid computing a collision detection if we already detected a collision.
+			if (hasCollided)
+				return;
+
+			otherRect = {otherTransformComponent.mPosition.x, otherTransformComponent.mPosition.y, otherCollisionComponent.mRectScale.x, otherCollisionComponent.mRectScale.y};
+			if (e != enemyEntity && CheckCollisionRecs(newEntityRect, otherRect))
+			{
+				hasCollided = true;
+			}
+		});
+
+
+		if (!hasCollided)
+		{
+			transformComponent.mPosition = newPosition;
+		}
+	}
```

Comment:
You probably noticed this is similar code for the player and the enemies, so this should be possible to extract that logic in one system that could work for both, that's the nice thing with ECS, we can share the logic between "different" types of entities as long as they have the required components.